File: seed/pydantic-v2/trace/resources/commons/types/list_type.py
from pydantic import BaseModel
from resources.commons.types.variable_type import VariableType
from typing import Optional


class ListType(BaseModel):
    value_type: VariableType = Field(alias="valueType")
    is_fixed_length: Optional[bool] = Field(alias="isFixedLength", default=None)
    """
    Whether this list is fixed-size (for languages that supports fixed-size lists). Defaults to false.
    """

File: seed/ruby-sdk-v2/trace/lib/fern_trace/commons/types/list_type.rb
# frozen_string_literal: true

module FernTrace
  module Commons
    module Types
      class ListType < Internal::Types::Model
        field :value_type, -> { FernTrace::Commons::Types::VariableType }, optional: false, nullable: false, api_name: "valueType"
        field :is_fixed_length, -> { Internal::Types::Boolean }, optional: true, nullable: false, api_name: "isFixedLength"
      end
    end
  end
end
